src/libraries/Microsoft.PowerFx.Interpreter/Environment/RuntimeConfig.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // This should have read-only views on all members. 
+    // Assume any field is nullable. 
+    public interface IRuntimeConfig
+    {
+        /// <summary>
+        /// This should match the SymbolTable provided at bind time. 
+        /// </summary>
+        public ReadOnlySymbolValues Values { get; }
+
+        /// <summary>
+        /// Services. 
+        /// </summary>
+        public IServiceProvider Services { get; }
+    }
+
+    /// <summary>
+    /// Runtime configuration for the execution of an expression.
+    /// This can be reused across multiple evals - so it shouldn't have any state that expires such as a cancellation token. 
+    /// </summary>
+    public sealed class RuntimeConfig : IRuntimeConfig
```

Comment:
Before, we could just pass a SymbolValues around because it had both Values and IServiceProvider. 
Now we need a new class to contain both, hence RuntimeConfig .